plugins/PrivacyManager/Model/DataSubjects.php

```diff
@@ -31,22 +31,47 @@ public function __construct(LogTablesProvider $logTablesProvider)
         $this->logTablesProvider = $logTablesProvider;
     }
 
+    private function getDistinctIdSitesInTable($tableName, $maxIdSite)
+    {
+        $tableName = Common::prefixTable($tableName);
+        $idSitesLogTable = Db::fetchAll('SELECT DISTINCT idsite FROM ' . $tableName);
+        $idSitesLogTable = array_column($idSitesLogTable, 'idsite');
+        $idSitesLogTable = array_map('intval', $idSitesLogTable);
+        $idSitesLogTable = array_filter($idSitesLogTable, function ($idSite) use ($maxIdSite) {
+            return !empty($idSite) && $idSite <= $maxIdSite;
+        });
+        return $idSitesLogTable;
+    }
+
     public function deleteDataSubjectsForDeletedSites($allExistingIdSites)
     {
         if (empty($allExistingIdSites)) {
             return array();
         }
 
+        $allExistingIdSites = array_map('intval', $allExistingIdSites);
+        $maxIdSite = max($allExistingIdSites);
         $results = [];
 
+        $idSitesLogVisit = $this->getDistinctIdSitesInTable('log_visit', $maxIdSite);
+        $idSitesLogVisitAction = $this->getDistinctIdSitesInTable('log_link_visit_action', $maxIdSite);
+        $idSitesLogConversion = $this->getDistinctIdSitesInTable('log_conversion', $maxIdSite);
+        $idSitesUsed = array_unique(array_merge($idSitesLogVisit, $idSitesLogVisitAction, $idSitesLogConversion));
+
+        $idSitesNoLongerExisting = array_diff($idSitesUsed, $allExistingIdSites);
```

Comment:
Possible edge case that I haven't thought through too much, but what if a site hasn't tracked any visits yet, and this runs. It will be treated as a deleted site, correct? If tracking is installed while this process is under way, will it be a problem?

Reply:
It shouldn't cause it would only return sites here that were tracked into and not exist in sites?

Reply:
If sites has ids, [1,2,3,4,5] and sites that are tracked has ids [1,2,3], then the array_diff for $idSitesNoLongerExisting would be [4,5], correct?

Reply:
Let me try and explain again (I might be seeing it wrong too). There are 5 sites total:

[1,2,3,4,5]

No site is deleted. Only three of the sites have tracked data, [1,2,3]. So when looking through log tables for idsites, we only get `[1,2,3]`.

In this situation, we will try to delete log data for [4,5], which would be ok since there isn't any log data, except possibly if tracking begins during the deletion process. Again, I'm not sure if this is a problem ultimately, just mentioning it.

Reply:
@diosmosis it would return `[]` not `[4,5]` see eg https://3v4l.org/Q4uci

array_diff returns all entries from the first array, that are not in the second array.

Reply:
I see, no issue then. 👍